codersdk/workspaceagents.go

```diff
@@ -289,8 +291,13 @@ func (c *Client) DialWorkspaceAgent(ctx context.Context, agentID uuid.UUID, opti
 				options.Logger.Debug(ctx, "failed to dial", slog.Error(err))
 				continue
 			}
-			sendNode, errChan := tailnet.ServeCoordinator(websocket.NetConn(ctx, ws, websocket.MessageBinary), func(node []*tailnet.Node) error {
-				return conn.UpdateNodes(node, false)
+			sendNode, errChan := tailnet.ServeCoordinator(websocket.NetConn(ctx, ws, websocket.MessageBinary), func(nodes []*tailnet.Node) error {
+				if len(nodes) != 1 {
+					options.Logger.Warn(ctx, "incorrect number of nodes returned from ServeCoordinator", slog.F("len", len(nodes)))
+					return nil
+				}
```

Comment:
Wait really? We can only get one node at a time? Why is it an array to begin with then?

Reply:
Yeah, I'm not sure why though. This is actually vestigal though so it can be removed